livechats/tests.py

```diff
@@ -0,0 +1,124 @@
+        get_response = self.client.get(f'/conversations/{self.base_chat.data["_id"]}/',
+                                       **{"HTTP_AUTHORIZATION": f"Bearer {not_participant_token}"})
+        self.assertEqual(get_response.status_code, 403)
+        self.assertEqual(get_response.data["message"], "You are not the participant of this live-chat")
+
+    def test_get_conversation_invalid_id(self):
+        get_response = self.client.get(f'/conversations/22222/',
+                                       **{"HTTP_AUTHORIZATION": f"Bearer {self.user_initiator_token}"})
+        self.assertEqual(get_response.status_code, 400)
+        self.assertEqual(get_response.data["message"], "Provided invalid chat id")
+
+    def test_get_conversation_wrong_chat(self):
+        get_response = self.client.get(f'/conversations/65e0b7565e55df16a396c444/',
+                                       **{"HTTP_AUTHORIZATION": f"Bearer {self.user_initiator_token}"})
+        self.assertEqual(get_response.status_code, 404)
+        self.assertEqual(get_response.data["message"], 'Conversation does not exist')
+
+    def test_user_conversation(self):
+        get_response = self.client.get(f'/conversations/',
+                                       **{"HTTP_AUTHORIZATION": f"Bearer {self.user_initiator_token}"})
+        self.assertEqual(get_response.status_code, 200)
+        self.assertEqual(len(get_response.data), 1)
+
+    def test_restart_conversation(self):
+        post_response = self.client.post(f'/conversations/restart/{self.base_chat.data["_id"]}/',
+                                         **{"HTTP_AUTHORIZATION": f"Bearer {self.user_initiator_token}"})
+
+        self.assertEqual(post_response.status_code, 200)  # should be False
+        self.assertEqual(post_response.data["message"], "Chat was restarted")
+
+    def test_restart_conversation_invalid_id(self):
+        post_response = self.client.post(f'/conversations/restart/2222222/',
+                                         **{"HTTP_AUTHORIZATION": f"Bearer {self.user_initiator_token}"})
+        self.assertEqual(post_response.status_code, 400)  # should be False
+        self.assertEqual(post_response.data["message"], "Provided invalid chat id")
+
+    def test_restart_conversation_not_participant(self):
+        not_participant_token = RefreshToken.for_user(self.user_recipient_two).access_token
+        post_response = self.client.post(f'/conversations/restart/{self.base_chat.data["_id"]}/',
+                                         **{"HTTP_AUTHORIZATION": f"Bearer {not_participant_token}"})
+        self.assertEqual(post_response.status_code, 403)
+        self.assertEqual(post_response.data["message"], "You are not the participant of this live-chat")
+
+    def tearDown(self):
+        self.mock_conversations.stop()
+        self.mock_redis.stop()
+
+    @classmethod
+    def tearDownClass(cls):
+        client = MongoClient('localhost', 27017)
+        client.drop_database('livechats')
```

Comment:
Here you are dropping a real database. Although it's on localhost and in this case, it will drop the local database of each developer in our team (who decides to run the unit tests), which they may use for manual testing, still, it might be inappropriate to destroy a real (even if local) database in unit tests.